bindings/pyroot/src/Pythonize.cxx

```diff
@@ -2273,11 +2273,14 @@ namespace {
       Py_XDECREF(cppname);
 
       std::string pprint;
-      std::stringstream calcPrintValue;
-      calcPrintValue << "*((std::string*)" << &pprint << ") = cling::printValue((" << className << "*)"
-                     << self->GetObject() << ");";
-      gInterpreter->Calc(calcPrintValue.str().c_str());
-      return PyROOT_PyUnicode_FromString(pprint.c_str());
+      void *myObj = self->GetObject();
+      std::stringstream ss;
+      ss << myObj;
+      std::string code =  "*((" + className + "*)" + ss.str() + ")";
+      pprint = gInterpreter->Evaluate(code.c_str());
+      std::string delim = pprint.substr(pprint.find("&") + 3);
```

Comment:
Please don't do string operations here to "swallow" the type. I actually think that it's worth while to print the C++ type with the value, also to signal that this is a C++ (and thus typed!) entity. I.e. let's adjust the test, not the code?